plugins/module_utils/network/iosxr/argspec/ospf_interfaces/ospf_interfaces.py

```diff
@@ -76,19 +76,27 @@ def __init__(self, **kwargs):
                             },
                         },
                         "authentication": {
+                            "no_log": False,
                             "type": "dict",
                             "options": {
                                 "message_digest": {
+                                    "no_log": False,
                                     "type": "dict",
-                                    "options": {"keychain": {"type": "str"}},
+                                    "options": {
+                                        "keychain": {
+                                            "type": "str",
+                                            "no_log": False,
+                                        }
+                                    },
                                 },
                                 "null_auth": {"type": "bool"},
                             },
                         },
                         "authentication_key": {
                             "type": "dict",
+                            "no_log": False,
                             "options": {
-                                "password": {"type": "str"},
+                                "password": {"type": "str", "no_log": False},
```

Comment:
Please hold on to merging the no_log changes till we take a final call.

cc @GomathiselviS

Reply:
Rest LGTM!

Reply:
@ashwini-mhatre As decided, we are masking all keys that may have sensitive information as values. Once that is handled, please feel free to merge the PR.